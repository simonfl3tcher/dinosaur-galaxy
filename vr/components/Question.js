// @flow

// React
import React, { Component } from 'react';
import { Text, View, VrButton } from 'react-vr';

// Libs
import { map } from 'lodash';
import PropTypes from 'prop-types';
import styles from '../styles/main';

export default class Question extends Component {
  render() {
    return (
      <View>
        {map(this.props.question.answers, (value, key) => {
          return (
            <VrButton key={key} onClick={() => this.props.pickAnswer(key)}>
              <Text
                style={{
                  fontSize: 0.2,
                  textAlign: 'center',
                  color: '#000',
                  backgroundColor: '#fff',
                  margin: 0.1,
                  transform: [{ translate: [0, 2, -5] }],
                }}
              >
                {value}
              </Text>
            </VrButton>
          );
        })}
        <Text style={styles.questionBlock}>
          {this.props.question.question}
        </Text>
      </View>
    );
  }
}

Question.propTypes = {
  question: PropTypes.object,
  pickAnswer: PropTypes.func,
};
