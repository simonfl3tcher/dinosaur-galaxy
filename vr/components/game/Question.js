import React, { Component } from 'react';
import { View, Model, VrButton, asset } from 'react-vr';
import { map } from 'lodash';
import Character from './Character';

export default class Question extends Component {
  props: {
    question: Object,
    pickAnswer: Function,
    nextQuestion: Function
  };

  nextQuestion() {
    this.props.nextQuestion();
  }

  render() {
    let i = 1;
    return (
      <View>
        {map(this.props.question.answers, (value: string, key: string) => {
          return (
            <Character
              key={value.character}
              index={i++}
              model={value.character}
              pickAnswer={() => this.props.pickAnswer(key)}
            />
          );
        })}
        <Model
          style={{
            transform: [{ translate: [2, -1.5, -5] }, { scale: 0.02 }],
          }}
          lit={true}
          source={{
            obj: asset(`characters/dino-q${this.props.question.id}.obj`),
            mtl: asset(`characters/dino-q${this.props.question.id}.mtl`),
          }}
        />
        <VrButton onClick={() => this.nextQuestion()}>
          <Model
            style={{
              transform: [{ translate: [2, -2, -5] }, { scale: 0.01 }],
            }}
            lit={true}
            source={{
              obj: asset('next/next.obj'),
              mtl: asset('next/next.mtl'),
            }}
          />
        </VrButton>
      </View>
    );
  }
}
