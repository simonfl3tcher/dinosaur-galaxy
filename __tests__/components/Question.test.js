import 'react-native';
import 'react-vr';
import React from 'react';
import Question from '../../vr/components/game/Question.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const question = {
    question: 'example?',
    answers: {
      a: 'a',
      b: 'b',
      c: 'b',
    },
    answer: 'a',
  };

  const tree = renderer.create(<Question question={question} />).toJSON();
  expect(tree).toMatchSnapshot();
});
