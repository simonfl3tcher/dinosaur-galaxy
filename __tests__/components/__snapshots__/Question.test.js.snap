// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    disabled={false}
    ignoreLongClick={false}
    onClick={[Function]}
    onEnter={[Function]}
    onExit={[Function]}
    onInput={[Function]}
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      isOnLayer={false}
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#000",
          "fontSize": 0.2,
          "margin": 0.1,
          "textAlign": "center",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                2,
                -5,
              ],
            },
          ],
        }
      }
    >
      a
    </Text>
  </View>
  <View
    disabled={false}
    ignoreLongClick={false}
    onClick={[Function]}
    onEnter={[Function]}
    onExit={[Function]}
    onInput={[Function]}
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      isOnLayer={false}
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#000",
          "fontSize": 0.2,
          "margin": 0.1,
          "textAlign": "center",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                2,
                -5,
              ],
            },
          ],
        }
      }
    >
      b
    </Text>
  </View>
  <View
    disabled={false}
    ignoreLongClick={false}
    onClick={[Function]}
    onEnter={[Function]}
    onExit={[Function]}
    onInput={[Function]}
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      isOnLayer={false}
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#000",
          "fontSize": 0.2,
          "margin": 0.1,
          "textAlign": "center",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                2,
                -5,
              ],
            },
          ],
        }
      }
    >
      b
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isOnLayer={false}
    style={
      Object {
        "backgroundColor": "#fff",
        "color": "#000",
        "fontSize": 0.5,
        "textAlign": "center",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              1,
              -5,
            ],
          },
        ],
      }
    }
  >
    example?
  </Text>
</View>
`;
